source/simulator/assembly.cc

```diff
@@ -1227,6 +1227,206 @@ namespace aspect
         }
     };
 
+    /**
+     * A class for the definition of functions that implement the
+     * non-linear system terms for the newton compressible or
+     * incompressible equations.
+     */
+    template <int dim>
+    class NewtonSystem : public aspect::internal::Assembly::Assemblers::AssemblerBase<dim>,
+      public SimulatorAccess<dim>
+    {
+      public:
+        void
+        local_assemble_newton_preconditioner (const double                                             pressure_scaling,
```

Comment:
I'd still call it `local_assemble_stokes_preconditioner`. The "Newton" part is already taken care of by the class name.